app/services/checkout_services/create_order.rb

```diff
@@ -0,0 +1,36 @@
+# frozen_string_literal: true
+
+module CheckoutServices
+  class CreateOrder
+    def call(user)
+      cart_items = user.shopping_cart.cart_items
+      if cart_items.empty?
+        OpenStruct.new({ success?: false,
+                         message: 'Your shopping cart is empty. Add some products to continue checkout' })
+      else
+        create_order(user, cart_items)
+      end
+    end
+
+    private
+
+    def create_order(user, cart_items)
+      order = Order.new(user_id: user.id)
+      move_products_to_order(order, cart_items)
+      order.total_price = order.order_items.inject(0) { |sum, item| sum + (item.product.price * item.quantity) }
```

Comment:
Może warto to przenieść do metody?

ten sam kod dla cart_items jest w shoping cart_presenterze -> Może warto go użyć? Lub wyprowadzić do jakiegoś serwisu, typu CalculatePrice -> I przekazywać cart_items lub order_items?

Reply:
Przerzuciłem to do serwisu :)